### ISSUE TITLE
Cannot downgrade npm once upgraded to 3.5.0

### ISSUE THREAD

**Issue by toddself** [npm/npm#10654]
running `npm i -g npm@2` (which finds 2.14.13 correctly) does not install and errors out with:
```
npm WARN In npm@2.14.13 replacing bundled version of npmlog with npmlog@2.0.0
npm WARN In npm@2.14.13 replacing bundled version of gauge with gauge@1.2.2
npm WARN In npm@2.14.13 replacing bundled version of are-we-there-yet with are-we-there-yet@1.0.4
npm WARN In npm@2.14.13 replacing bundled version of delegates with delegates@0.1.0
npm WARN In npm@2.14.13 replacing bundled version of lodash.padleft with lodash.padleft@3.1.1
npm WARN In npm@2.14.13 replacing bundled version of has-unicode with has-unicode@1.0.1
npm WARN In npm@2.14.13 replacing bundled version of lodash.padright with lodash.padright@3.1.1
npm WARN In npm@2.14.13 replacing bundled version of lodash.pad with lodash.pad@3.1.1
npm WARN In npm@2.14.13 replacing bundled version of lodash._createpadding with lodash._createpadding@3.6.1
npm WARN In npm@2.14.13 replacing bundled version of lodash._basetostring with lodash._basetostring@3.0.1
npm WARN In npm@2.14.13 replacing bundled version of lodash.repeat with lodash.repeat@3.0.1
npm ERR! Darwin 15.0.0
npm ERR! argv "/opt/node-v4.2.1-darwin-x64/bin/node" "/opt/iojs-current/bin/npm" "i" "-g" "npm@2"
npm ERR! node v4.2.1
npm ERR! npm  v3.5.0
npm ERR! path /opt/node-v4.2.1-darwin-x64/lib/node_modules/npm/node_modules/fstream-npm/node_modules/fstream-ignore/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename

npm ERR! enoent ENOENT: no such file or directory, rename '/opt/node-v4.2.1-darwin-x64/lib/node_modules/npm/node_modules/fstream-npm/node_modules/fstream-ignore/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match' -> '/opt/node-v4.2.1-darwin-x64/lib/node_modules/npm/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match'
npm ERR! enoent ENOENT: no such file or directory, rename '/opt/node-v4.2.1-darwin-x64/lib/node_modules/npm/node_modules/fstream-npm/node_modules/fstream-ignore/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match' -> '/opt/node-v4.2.1-darwin-x64/lib/node_modules/npm/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/todd/src/scripto2/npm-debug.log
npm ERR! code 1
```

**Comment by othiym23** [comment]
This is a regression introduced most likely by the changes made to `bundleDependency` handling in `npm@3.5.0`. I was hopeful that the the fix that @iarna was working on for a related issue in the https://github.com/npm/npm/tree/iarna/fix-move-modules branch would fix this issue as well, but it doesn't. The interaction between `npm@3`'s bundling and deduping code is quite subtle and we're in the process of solidifying it. Thanks for filing this!

**Comment by othiym23** [comment]
Nevermind, I made a git booboo – @iarna and I have verified that `iarna/fix-move-modules` addresses this problem, and we'll try to get the fix included in this week's `npm@3` release.

**Comment by toddself** [comment]
a+ thank you!

**Comment by toddself** [comment]
This is fixed in at least 3.9.3 and 3.10.2!

Hadn't thought to double-check for a while.  Thank you!!!
